openwisp_controller/config/static/config/js/widget.js

```diff
@@ -248,6 +298,13 @@
                 $(e.target).val('');
             }
         });
+
+        // fill device context field with default values of selected templates.
+        getDefaultValues();
```

Comment:
here you could supply a parameter that tells the function to also updated `django._njc_initial_values['config-0-context']` so the unsaved changes bug should disappear.

Reply:
@nemesisdesign I have noticed this `unsaved_change` on the master branch also. But I have included the changes requested though.